src/workers/cdc/common.ts

```diff
@@ -8,3 +24,90 @@ export const getTableName = <Entity>(
   con: DataSource,
   target: EntityTarget<Entity>,
 ): string => con.getRepository(target).metadata.tableName;
+
+export const notifyPostContentUpdated = async ({
+  con,
+  post,
+}: {
+  con: DataSource;
+  post: ChangeObject<Post>;
+}): Promise<void> => {
+  const [source, keywords, relatedPosts] = await Promise.all([
+    con.getRepository(Source).findOneBy({
+      id: post.sourceId,
+    }),
+    con.getRepository(PostKeyword).find({
+      where: {
+        postId: post.id,
+      },
+      order: {
+        keyword: 'ASC',
+      },
+    }),
+    con.getRepository(PostRelation).find({
+      where: [
+        {
+          postId: post.id,
+        },
+        {
+          relatedPostId: post.id,
+        },
+      ],
+      order: {
+        createdAt: 'ASC',
+      },
+    }),
+  ]);
+  const articlePost = post as ChangeObject<ArticlePost>;
+  const freeformPost = post as ChangeObject<FreeformPost>;
+
+  const contentUpdatedMessage = new ContentUpdatedMessage({
+    yggdrasilId: post.yggdrasilId,
+    postId: post.id,
+    type: post.type,
+    title: post.title,
+    createdAt: post.createdAt,
+    updatedAt: post.metadataChangedAt,
+    source: source
+      ? {
+          ...source,
+          createdAt: +source.createdAt,
+        }
+      : undefined,
+    tags: post.tagsStr?.split(',') || [],
+    keywords: keywords.map((item) => item.keyword),
+    banned: post.banned,
+    private: post.private,
+    visible: post.visible,
+    origin: post.origin,
+    url: articlePost.url,
+    canonicalUrl: articlePost.canonicalUrl,
+    image: articlePost.image,
+    description: articlePost.description,
+    readTime: articlePost.readTime,
+    summary: articlePost.summary,
+    content: freeformPost.content,
+    language: post.language,
+    contentMeta: post.contentMeta
```

Comment:
@denisb0 is https://github.com/dailydotdev/schema/blob/main/daily-api/posts.proto#L33 `int64` for some specific reason or?

Reply:
@vpol do you maybe know if we need that big of a numbers, we usually use int32 in other schemas and on JS side it is a bit tricky because int64 has to be [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) which is more complex to serialize to JSON.

Reply:
If basically defines a vector length, event int16 will do the job.